Algorithm.CSharp/CoarseUniverseSplitRegressionAlgorithm.cs

```diff
@@ -0,0 +1,110 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This is a regression algorithm to test the price and split factors in coarse fundamental data
+    /// </summary>
+    /// <meta name="tag" content="using data" />
+    /// <meta name="tag" content="universes" />
+    /// <meta name="tag" content="coarse universes" />
+    /// <meta name="tag" content="regression test" />
+    public class CoarseUniverseSplitRegressionAlgorithm : QCAlgorithm
```

Comment:
Wondering if a full regression algorithm is a little heavy here. It seems most of this test is confirming that we're able to parse out these new values and could be written as a unit test against the `CoarseFundamental` type using string literals for input lines to be parsed. The nice part about going the unit test route is that they're super fast and fail travis.

thoughts?